src/core.js

```diff
@@ -2059,26 +2059,8 @@ define([
     };
 
     function getWidgetClassAndOptions(propPath, mug) {
-        var propDef = mug.p.getDefinition(propPath),
-            propVal = mug.p[propPath];
-
-        // handle properties whose visibility depends on other properties'
-        // visibility (implied visible_if_present on the depending property
-        // takes precedence)
-        if (_.isUndefined(propVal) &&
-            propDef && propDef.visibility &&
-            mug.p.getDefinition(propDef.visibility) &&
-            !getWidgetClassAndOptions(propDef.visibility, mug))
```

Comment:
That `_.isUndefined(propVal)` up there is causing me some consternation. Dependent visibility was ignored unless the property value was undefined... so strange.